### ISSUE TITLE
Server died after a few hours of running

### ISSUE THREAD

**Issue by WardBrian** [flatironinstitute/mcmc-monitor#1]
Here is the end of the output
```
Webrtc getSequences 36
webrtc peer disconnected
/mnt/ceph/users/bward/mcmc/node_modules/simple-peer/index.js:248
    if (this.destroyed) throw errCode(new Error('cannot send after peer is destroyed'), 'ERR_DESTROYED')
                                      ^

Error: cannot send after peer is destroyed
    at Peer.send (/mnt/ceph/users/bward/mcmc/node_modules/simple-peer/index.js:248:39)
    at /mnt/ceph/users/bward/mcmc/node_modules/mcmc-monitor/dist/Server.js:109:30 {
  code: 'ERR_DESTROYED'
}

Node.js v18.12.1
```

**Comment by magland** [comment]
@jsoules

Seems that a peer.send() was attempted after the webrtc connection was lost.

The code that handles the WebRTC peer communication via the simplepeer library is all contained here:

https://github.com/flatironinstitute/mcmc-monitor/blob/459e8b6814c745bbf7681f3035b409177a540057/service/src/Server.ts#L52-L88

We'll want to do a few things to make this more robust.

1. Extract this code block out into it's own file/class
2. Wrap all the simplepeer calls in try/catch blocks. That includes peer.destroy(), peer.send(), peer.signal()
3. Be more careful about managing connections - when the webrtc connection is lost, what steps are taken to recover? How to we avoid establishing more than one connection, etc. We'll want to examine the SignalCommunicator class as part of this.

As far as I can guess, what happened here is that the webrtc connection happened to get lost during the async processing of a request (handleApiRequest). Then peer.send() was called on an already closed connection. Wrapping this in try/catch and writing the error as a warning to the console would solve this problem.

For now I think we should address items 1 and 2. Number 3 will require some more thought.

**Comment by magland** [comment]
Rather than rely solely on try/catch, I think it's better to check whether the connection is still alive here (after the request is processed):

https://github.com/flatironinstitute/mcmc-monitor/blob/459e8b6814c745bbf7681f3035b409177a540057/service/src/Server.ts#L61-L68

(we should do try/catch/warn also)

We should assume that the connection object lives as long as the connection is active. When it disconnects, we'll just wait until a new connection object is obtained. In other words, the connection object is not going to become reconnected once the connection is lost.

**Comment by magland** [comment]
Fixed by #7 and included in latest version on npm (0.1.9)
